### PR TITLE
feat: Admin debug mode warning

### PR DESCRIPTION
**Pull request by imorland**
**Changes proposed in this pull request:**
Adds a `debug` warning widget to the admin dashboard. When `debug` is not enabled, the widget is not displayed.

**Reviewers should focus on:**
How's the `less`? I'm terrible...

**Screenshot**
image

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

File: framework/core/js/src/admin/components/DebugWarningWidget.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  content() {
+    return (
+      <div>
+        <h4 className="DebugWarningWidget-title">
+          {icon('fas fa-exclamation-triangle')} {app.translator.trans('core.admin.debug-warning.label')}
+        </h4>
+        <p className="DebugWarningWidget-detail">{app.translator.trans('core.admin.debug-warning.detail')}</p>
+      </div>
+    );
+  }
```
Comment (SychO9) [inline framework/core/js/src/admin/components/DebugWarningWidget.js]: We can probably use the `Alert` component directly here and then make any necessary CSS tweaks.

Comment (imorland) [reply framework/core/js/src/admin/components/DebugWarningWidget.js]: I think I follow.. updated, that better?

**Comment by SychO9** [comment]
I pushed what I had in mind would be a good opportunity to further improve the alert component so that it can be used in similar ways without hassle or custom style changes. This is the result (`error` type alert too much?)

Screenshot from 2022-08-11 18-21-14

**Comment by imorland** [comment]
Was just looking at this, I think the changes to the `Alert` component are useful 👍 

Only thought is, for _this_ use case, I think it should be a `warning`, rather than `error`?

**Comment by SychO9** [comment]
> Was just looking at this, I think the changes to the `Alert` component are useful +1
> 
> Only thought is, for _this_ use case, I think it should be a `warning`, rather than `error`?

agreed 🙈

File: framework/core/js/src/admin/components/DebugWarningWidget.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export default class DebugWarningWidget extends DashboardWidget {
+  className() {
+    return 'DebugWarningWidget';
+  }
+
+  content() {
+    return (
+      <Alert
+        type="error"
```
Comment (SychO9) [inline framework/core/js/src/admin/components/DebugWarningWidget.js]:
```suggestion
        type="warning"
```

File: framework/core/locale/core.yml [hunk 3]
```diff
@@ -59,6 +59,14 @@ core:
       title: Dashboard
       tools_button: Tools
 
+    # These translations are usin in the debug warning widget.
+    debug-warning:
+      detail: |
+        When <code>debug</code> mode is active, Flarum will rebuild it's <code>js</code> and <code>less</code> assets on every request, and could also potentially leak other information, such as database secrets, environment variables, etc.
+
+        It is highly recommended to disable <code>debug</code> in your <code>config.php</code> file when debugging is not required.
+      label: Debug mode active
```
Comment (SychO9) [inline framework/core/locale/core.yml]:
```suggestion
    debug-warning:
      detail: |
        When <code>debug</code> mode is active, Flarum will rebuild its <code>JavaScript</code> and <code>CSS</code> assets on every request, and could also potentially leak other information, such as database secrets, environment variables, etc.

        It is highly recommended to disable <code>debug</code> in your <code>config.php</code> file in production.
      label: Debug mode active
```

**Comment by imorland** [comment]
Updated screenshot with latest changes

**Review by SychO9 — APPROVED** [review]

File: framework/core/js/src/common/components/Alert.tsx [hunk 4]
```diff
@@ -51,6 +58,12 @@ export default class Alert<T extends AlertAttrs = AlertAttrs> extends Component<
 
     return (
       <div {...attrs}>
+        {title ? (
+          <div class="Alert-title">
+            {icon ? <span className="Alert-title-icon">{iconHelper(icon)}</span> : null}
+            <span className="Alert-title-text">{title}</span>
+          </div>
+        ) : null}
```
Comment (davwheat) [inline framework/core/js/src/common/components/Alert.tsx]:
```suggestion
        {!!title && (
          <div class="Alert-title">
            {!!icon && <span class="Alert-title-icon">{iconHelper(icon)}</span>}
            <span class="Alert-title-text">{title}</span>
          </div>
        )}
```

File: framework/core/locale/core.yml [hunk 5]
```diff
@@ -59,6 +59,14 @@ core:
       title: Dashboard
       tools_button: Tools
 
+    # These translations are usin in the debug warning widget.
+    debug-warning:
+      detail: |
+        When <code>debug</code> mode is active, Flarum will rebuild its <code>JavaScript</code> and <code>CSS</code> assets on every request, and could also potentially leak other information, such as database secrets, environment variables, etc.
+
+        It is highly recommended to disable <code>debug</code> in your <code>config.php</code> file in production.
```
Comment (davwheat) [inline framework/core/locale/core.yml]: Do we have a docs page we could link to about this, just for clarification?

**Review by davwheat — CHANGES_REQUESTED** [review]

File: framework/core/locale/core.yml (hunk 5 above)
Comment (imorland) [reply framework/core/locale/core.yml]: We have https://docs.flarum.org/troubleshoot/#step-0-activate-debug-mode

Perhaps this is what you had in mind?

Comment (davwheat) [reply framework/core/locale/core.yml]: Yeah, I think that works.

**Review by davwheat — APPROVED** [review]

**Pull request merged**